src/OrleansRuntime/Startup/SiloHostBuilder.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Host;
+
+namespace Orleans.Runtime.Startup
+{
+    public class SiloHostBuilder : ISiloHostBuilder
+    {
+        private ClusterConfiguration clusterConfiguration;
+        private readonly List<Action<IServiceCollection>> configureServicesDelegates = new List<Action<IServiceCollection>>();
```

Comment:
What's the reason you want to have a list of configureServiceDelegate, wouldn't one be enough?